src/PixiEditor.ChangeableDocument/Changes/Drawing/ShiftLayer_UpdateableChange.cs

```diff
@@ -28,48 +43,57 @@ public void Update(VecI delta, bool keepOriginal)
         this.keepOriginal = keepOriginal;
     }
 
-    private HashSet<VecI> DrawShiftedLayer(Document target)
-    {
-        var targetImage = target.FindMemberOrThrow<Layer>(layerGuid).LayerImage;
-        var prevChunks = targetImage.FindAffectedChunks();
-        targetImage.CancelChanges();
-        if (!keepOriginal)
-            targetImage.EnqueueClear();
-        targetImage.EnqueueDrawChunkyImage(delta, targetImage, false, false);
-        var curChunks = targetImage.FindAffectedChunks();
-        curChunks.UnionWith(prevChunks);
-        return curChunks;
-    }
-
     public override OneOf<None, IChangeInfo, List<IChangeInfo>> Apply(Document target, bool firstApply, out bool ignoreInUndo)
     {
-        var chunks = DrawShiftedLayer(target);
-        var image = target.FindMemberOrThrow<Layer>(layerGuid).LayerImage;
-
-        if (originalLayerChunks is not null)
-            throw new InvalidOperationException("saved chunks is not null even though we are trying to save them again");
-        originalLayerChunks = new(image, image.FindAffectedChunks());
-        image.CommitChanges();
+        originalLayerChunks = new Dictionary<Guid, CommittedChunkStorage?>();
+        List<IChangeInfo> changes = new List<IChangeInfo>();
+        foreach (var layerGuid in layerGuids)
+        {
+            var chunks = ShiftLayerHelper.DrawShiftedLayer(target, layerGuid, keepOriginal, delta);
+            var image = target.FindMemberOrThrow<Layer>(layerGuid).LayerImage;
+            
+            changes.Add(new LayerImageChunks_ChangeInfo(layerGuid, chunks));
+            
+            originalLayerChunks[layerGuid] = new(image, image.FindAffectedChunks());
+            image.CommitChanges();
+        }
 
         ignoreInUndo = delta.TaxicabLength == 0;
-        return new LayerImageChunks_ChangeInfo(layerGuid, chunks);
+        return changes;
     }
 
     public override OneOf<None, IChangeInfo, List<IChangeInfo>> ApplyTemporarily(Document target)
     {
-        var chunks = DrawShiftedLayer(target);
-        return new LayerImageChunks_ChangeInfo(layerGuid, chunks);
+        _tempChanges.Clear();
+
+        foreach (var layerGuid in layerGuids)
+        {
+            var chunks = ShiftLayerHelper.DrawShiftedLayer(target, layerGuid, keepOriginal, delta);
+            _tempChanges.Add(new LayerImageChunks_ChangeInfo(layerGuid, chunks));
+        }
+        
+        return _tempChanges;
```

Comment:
looks sus, why keep a reference to a returned list locally? who knows what it's gonna be used for by the caller. Also why store it at all?